crates/aptos/src/common/supra/mod.rs

```diff
@@ -0,0 +1,18 @@
+// Copyright © Aptos Foundation
+
+use crate::common::types::{CliTypedResult, TransactionOptions};
+use aptos_types::transaction::TransactionPayload;
+
+/// Arguments required by supra cli for its operation.
+pub struct SupraCommandArguments {
+    pub payload: TransactionPayload,
+    pub profile: Option<String>,
+    pub rpc: Option<reqwest::Url>,
+}
+
+/// Trait required by supra cli for its operation.
+pub trait SupraCommand {
+
+    /// consume self and returns [SupraCommandArguments]
```

Comment:
This is a conversion function, and I think it is not necessary to consume itself, which will make it flexible in the future.

Reply:
Hi simon, as the default behaviour of aptos cli consume self, hence most of the structs do not implement clone and let the execute function takes ownership. therefore the conversion function is not able to take &self because of clone issue.